homeassistant/components/powerwall/config_flow.py

```diff
@@ -257,6 +257,8 @@ async def async_step_reauth_confirm(
                 {CONF_IP_ADDRESS: reauth_entry.data[CONF_IP_ADDRESS], **user_input}
             )
             if not errors:
+                # We have a new valid connection, old cookie is no longer valid
+                user_input[CONFIG_ENTRY_COOKIE] = None
                 return self.async_update_reload_and_abort(
                     reauth_entry, data_updates=user_input
```

Comment:
```suggestion
                    reauth_entry, data_updates={**user_input, CONFIG_ENTRY_COOKIE: None}
```
This would be a bit cleaner than mutate user_input